### DISCUSSION TITLE
streamResponse.js  sendMessage socket closed unexpected

### DISCUSSION THREAD

**Discussion by hc0405** [danny-avila/LibreChat#1478] [Troubleshooting]
### What happened?

When the client displays the message, it suddenly stops.
I added a log in the streamResponse.js of the server.
The socket suddenly disconnects after sending half of the text.

let ok = res.write(`event: ${event}\ndata: ${JSON.stringify(message)}\n\n`);
  if (!ok) {
    logger.warn(
      `Socket closed:${res.closed} status:${res.statusCode} finished:${res.finished} writable:${res.writable} writableLength:${res.writableLength}  destroyed:${res.destroyed} writableFinished:${res.writableFinished}`,
    );
  }

### Steps to Reproduce

1. use docker install in GCE 
2.  use openai gpt-4 0613 
3. ask '測試與驗證'

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
warn: Socket closed:true status:200 finished:false writable:true writableLength:0  destroyed:true writableFinished:false
```

### Screenshots

<img width="820" alt="截圖 2024-01-03 下午2 52 03" src="https://github.com/danny-avila/LibreChat/assets/5293959/ed0188ef-ddd8-423e-8c33-d0a3e97dc45a">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I don't experience this issue. This is likely a network issue in your case. I noticed that it's rather slow with GPT-4 specifically, so you could be experiencing a timeout from the OpenAI servers as well

Ran with no errors:
![image](https://github.com/danny-avila/LibreChat/assets/110412045/bf9a530e-5094-4287-b65d-1ebbcfbbbb60)

**Reply by hc0405** [reply]
Everything works normally when I run it on localhost. However, as soon as I deploy it on GCE and run it remotely, this issue always arises. When it happens, refreshing the page displays all the content.

Also, as I observe the logs, OpenAI is responding with normal results. The problem occurs mid-way when sending them to the client side, causing a disconnection. I wonder if there's a chance to implement a reconnection mechanism, or perhaps extend the timeout period?
